test/test-0cf405b0.c

```diff
@@ -1,12 +1,13 @@
 #include <unistd.h> /* execlp(2), */
-#include <stdlib.h> /* exit(3), */
+#include <stdlib.h> /* exit(3), getenv(3), setenv(3)*/
 #include <string.h> /* strcmp(3), */
 
 int main(int argc, char *argv[])
 {
-	if (argc == 0) //strcmp(argv[0], "/proc/self/exe") == 0)
+	if (getenv("PROC_SELF_EXE") != NULL)
```

Comment:
Where is `PROC_SELF_EXE` being defined?

Reply:
Below on line 10.
When the program is executed the check fails leading to the recursive invocation via /process/self/exe that sets the environment variable before running exec.